src/operator/contrib/rroi_align.cc

```diff
@@ -0,0 +1,316 @@
+
+  MSHADOW_REAL_TYPE_SWITCH(in_data[0].type_flag_, DType, {
+    RROIAlignForward<DType>(ctx, param, in_data, req, out_data);
+  })
+}
+
+DMLC_REGISTER_PARAMETER(RROIAlignParam);
+
+NNVM_REGISTER_OP(_contrib_RROIAlign)
+.describe(R"code(Performs Rotated ROI Align on the input array.
+
+This operator takes a 4D feature map as an input array and region proposals as `rois`,
+then align the feature map over sub-regions of input and produces a fixed-sized output array.
+
+Different from ROI Align, RROI Align uses rotated rois, which is suitable for text detection.
+RRoIAlign computes the value of each sampling point by bilinear interpolation from the nearby
+grid points on the rotated feature map. No quantization is performed on any coordinates
+involved in the RoI, its bins, or the sampling points. Bilinear interpolation is used to
+compute the exact values of the input features at four regularly sampled locations in
+each RoI bin. Then the feature map can be aggregated by avgpooling.
+
+References
+----------
+
+Ma, Jianqi, et al. "Arbitrary-Oriented Scene Text Detection via Rotation Proposals."
+IEEE Transactions on Multimedia, 2018.
+
+)code" ADD_FILELINE)
+.set_num_inputs(2)
+.set_num_outputs(1)
+.set_attr<nnvm::FListInputNames>("FListInputNames",
+    [](const NodeAttrs& attrs) {
+  return std::vector<std::string>{"data", "rois"};
+})
+.set_attr<nnvm::FListOutputNames>("FListOutputNames",
+    [](const NodeAttrs& attrs) {
+  return std::vector<std::string>{"output"};
+})
+.set_attr_parser(ParamParser<RROIAlignParam>)
+.set_attr<mxnet::FInferShape>("FInferShape", [](const nnvm::NodeAttrs& attrs,
+      mxnet::ShapeVector *in_shape, mxnet::ShapeVector *out_shape){
+  using namespace mshadow;
+  const RROIAlignParam& param = nnvm::get<RROIAlignParam>(attrs.parsed);
+  CHECK_EQ(in_shape->size(), 2U) << "Input:[data, rois]";
+  // data: [batch_size, c, h, w]
+  mxnet::TShape dshape = in_shape->at(rroialign::kData);
+  CHECK_EQ(dshape.ndim(), 4U) << "data should be a 4D tensor";
+  // bbox: [num_rois, 6]
+  mxnet::TShape bshape = in_shape->at(rroialign::kBox);
+  CHECK_EQ(bshape.ndim(), 2U) << "bbox should be a 2D tensor of shape [batch, 6]";
+  CHECK_EQ(bshape[1], 6U) << "bbox should be a 2D tensor of shape [batch, 6]";
+  // out: [num_rois, c, pooled_h, pooled_w]
+  out_shape->clear();
+  out_shape->push_back(Shape4(bshape[0], dshape[1], param.pooled_size[0], param.pooled_size[1]));
+  return true;
+})
+.set_attr<nnvm::FInferType>("FInferType", [](const nnvm::NodeAttrs& attrs,
+      std::vector<int> *in_type, std::vector<int> *out_type) {
+  CHECK_EQ(in_type->size(), 2U);
+  int dtype = (*in_type)[0];
+  CHECK_EQ(dtype, (*in_type)[1]);
+  CHECK_NE(dtype, -1) << "Input must have specified type";
+
+  out_type->clear();
+  out_type->push_back(dtype);
+  return true;
+})
+.set_attr<FCompute>("FCompute<cpu>", RROIAlignForwardCompute<cpu>)
+.set_attr<nnvm::FGradient>("FGradient", MakeZeroGradNodes)
```

Comment:
Thanks for your contribution! 
We can define a backard function, which raises NotImplemented exception by LOG(FATAL).